### PR TITLE
Make button components work on preview.roadie.io

### PR DESCRIPTION
**Pull request by dtuite**

### CODE REVIEW

File: gatsby-config.js [hunk 1]
```diff
@@ -53,12 +59,13 @@ const getContentfulOptions = () => {
...
 // branch of a split testing experiment we are on.
 process.env.GATSBY_GIT_BRANCH_NAME = process.env.BRANCH;
+process.env.GATSBY_NETLIFY_SITE_NAME = process.env.SITE_NAME;
 
 module.exports = {
   siteMetadata: {
     title: SITE_TITLE,
     description: 'Hosted, managed, enterprise Backstage',
-    siteUrl: 'https://roadie.io',
+    siteUrl: getSiteUrl(),
```
Comment (dtuite) [inline gatsby-config.js:68]: This is not directly related to this fix but seems sensible anyway. On the preview site, the site URl is not roadie.io.

File: gatsby-config.js [hunk 2]
```diff
@@ -134,6 +141,11 @@ module.exports = {
         theme_color: theme.COLORS_PRIMARY_500,
         display: `minimal-ui`,
         icon: 'content/assets/logos/roadie/roadie-r-round.svg',
+        // The request for the manifest was failing with a 401 on preview.roadie.io because
+        // of the basic authentication layer applpied to that site by Netlify. We use basic
+        // auth on the preview site to keep it hidden from search engines and (to a less
+        // important extent) people.
+        crossOrigin: `use-credentials`,
```
Comment (dtuite) [inline gatsby-config.js:148]: This solves one of the 401s we encounter on preview.roadie.io. Explained here: https://thatemil.com/blog/2018/02/21/pwa-basic-auth/

File: src/templates/ChangelogList.js [hunk 3]
```diff
@@ -4,6 +4,17 @@ import { SEO, TextLink as Link, SitewideHeader, SitewideFooter } from 'component
...
+// In the code below, this is used to prevent the changelog from being collapsed on the preview
+// site. There is basic authentication on the preview site (preview.roadie.io) to prevent it being
+// indexed by search engines. This basic authentication is preventing prefetching from working,
+// because the requests to prefetch content do not have the authentication information added.
+// For some reason I (David) cannot understand, the ChangeSet componentis being chunked and
+// prefetched. On the preview site, it never gets loaded because of the prefetch authentication
+// problem, and I cannot expand the changelog items to check that the content looks as expected.
+// For this reason, I am using an env var to toggle off collapsing on the preview site but
+// nowhere else.
+import { isPreviewSite } from '../environment';
```
Comment (dtuite) [inline src/templates/ChangelogList.js:16]: The hack is explained in the comment. On preview.roadie.io, the following resources 401:

Screenshot 2022-03-14 at 23 44 19

This correlates with the prefetch items in the head.

Screenshot 2022-03-14 at 23 44 47

I read many docs but I couldn't come up with a way to make these resources be included in the initial bundle.

File: src/templates/ChangelogList.js [hunk 4]
```diff
@@ -46,7 +57,7 @@ const Changelog = ({
             releasedAt={releasedAt}
             description={description && description.childMarkdownRemark}
             slug={slug}
-            isCollapsible={true}
+            isCollapsible={isPreviewSite() ? false : true}
```
Comment (dtuite) [inline src/templates/ChangelogList.js:60]: So this is my hack

File: src/templates/ChangelogList.js (hunk 3 above)
Comment (Xantier) [reply src/templates/ChangelogList.js:16]: Are we able to turn off prefetching globally for the preview site with this: https://www.gatsbyjs.com/docs/reference/config-files/gatsby-browser/#disableCorePrefetching?

**Review by Xantier — APPROVED** [review]
Is grand anyways, a hack is better than those errors

File: src/templates/ChangelogList.js (hunk 3 above)
Comment (dtuite) [reply src/templates/ChangelogList.js:16]: I don't fully understand the docs but that seems to only be available inside Gatsby plugins. I've seen a few GitHub issues where people are asking how to turn off prefetching and Gatsby support basically say that you can't.

**Comment by dtuite** [comment]
Needs another quick look. I tried a path of using `<a>` instead of `<button>` but it didn't make a difference.

**Review by Xantier — APPROVED** [review]
